src/test/java/com/rocketden/main/api/RoomTests.java

```diff
@@ -425,51 +372,116 @@ public void changeRoomHostFailure() throws Exception {
+
+        UserDto user = new UserDto();
+        user.setNickname("User");
+
+        RoomDto room = setUpRoomWithTwoUsers(host, user);
+
         // New host does not exist in the room
-        updateHostRequest = new UpdateHostRequest();
+        UpdateHostRequest updateHostRequest = new UpdateHostRequest();
         updateHostRequest.setInitiator(host);
         updateHostRequest.setNewHost(new UserDto());
 
-        ERROR = UserError.NOT_FOUND;
+        ApiError ERROR = UserError.NOT_FOUND;
 
-        result = this.mockMvc.perform(put(String.format(PUT_ROOM_HOST, room.getRoomId()))
+        MvcResult result = this.mockMvc.perform(put(String.format(PUT_ROOM_HOST, room.getRoomId()))
                 .contentType(MediaType.APPLICATION_JSON_VALUE)
                 .content(Utility.convertObjectToJsonString(updateHostRequest)))
                 .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
                 .andReturn();
 
-        jsonResponse = result.getResponse().getContentAsString();
-        actual = Utility.toObject(jsonResponse, ApiErrorResponse.class);
+        String jsonResponse = result.getResponse().getContentAsString();
+        ApiErrorResponse actual = Utility.toObject(jsonResponse, ApiErrorResponse.class);
 
         assertEquals(ERROR.getResponse(), actual);
     }
+
+    /**
+     * Helper method that creates a room with two users
+     * @param host the host of the room
+     * @param user the second user who joins the room
+     * @return the resulting RoomDto object
+     * @throws Exception any error that occurs
+     */
+    private RoomDto setUpRoomWithTwoUsers(UserDto host, UserDto user) throws Exception {
+        // First, create the room
+        CreateRoomRequest createRequest = new CreateRoomRequest();
+        createRequest.setHost(host);
+
+        MvcResult result = this.mockMvc.perform(post(POST_ROOM)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(createRequest)))
+                .andDo(print()).andExpect(status().isCreated())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        RoomDto room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        // A second user joins the room
+        JoinRoomRequest joinRequest = new JoinRoomRequest();
+        joinRequest.setRoomId(room.getRoomId());
+        joinRequest.setUser(user);
+
+        result = this.mockMvc.perform(put(PUT_ROOM)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(joinRequest)))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        jsonResponse = result.getResponse().getContentAsString();
+        room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        assertEquals(host, room.getHost());
+        assertEquals(2, room.getUsers().size());
```

Comment:
nit, personal preference: Verify that the non-host user is indeed the one that you added.

Reply:
Added a contains statement to check if the list contains the UserDto that we specified